src/EnergyPlus/CrossVentMgr.cc

```diff
@@ -166,181 +164,168 @@ namespace CrossVentMgr {
         state.dataCrossVentMgr->HA_R = 0.0;
 
         // Is the air flow model for this zone set to UCSDCV Cross Ventilation?
-        if (state.dataRoomAirMod->IsZoneCV(ZoneNum)) {
+        if (state.dataRoomAir->IsZoneCrossVent(ZoneNum)) {
+
+            Real64 zoneJetRecAreaRatio = state.dataRoomAir->JetRecAreaRatio(ZoneNum);
+                
             // WALL Hc, HA and HAT calculation
-            for (int Ctd = state.dataUCSDShared->PosZ_Wall((ZoneNum - 1) * 2 + 1); Ctd <= state.dataUCSDShared->PosZ_Wall((ZoneNum - 1) * 2 + 2);
-                 ++Ctd) {
-                int SurfNum = state.dataUCSDShared->APos_Wall(Ctd);
+            for (int Ctd = state.dataRoomAir->PosZ_Wall(ZoneNum).beg; Ctd <= state.dataRoomAir->PosZ_Wall(ZoneNum).end; ++Ctd) {
```

Comment:
I cannot tell you how much I hate the pattern of faking what should be an array of structs or even multiple arrays in a single array by, e.g., putting one of the elements in the odd positions in the array and another element in the even positions. It's not cute. It's brittle and opaque and evil.